### PR TITLE
Add view category pages for donations

### PR DESCRIPTION
**Pull request by jamessspanggg**
- Add donations view category page
- Add view all donations page

### CODE REVIEW

**Comment by jamessspanggg** [comment]
Waiting for #114 to be merged first

File: utils/api/time.js [hunk 1]
```diff
@@ -4,3 +4,7 @@ export const getTimeDifferenceFromNow = (timeInMilliSeconds) => {
   let momentTime = moment.unix(timeInMilliSeconds / 1000);
   return momentTime.fromNow();
 };
+
+export const getFormattedDate = (timeInMilliSeconds) => {
+  return moment.unix(timeInMilliSeconds / 1000).format('DD/MM/YYYY');
```
[the remark this answers is not in the document]
Comment (jamessspanggg) [reply utils/api/time.js]: yes can, have updated accordingly

File: pages/donations/category/[categoryId].js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import SessionProvider from '../../../src/components/session/modules/SessionProvider';
+import { isAuthenticated } from '../../../utils/authentication/authentication';
+import Error from 'next/error';
+const TopNavigationBar = dynamic(() => import('../../../src/components/navbar/modules/TopNavigationBar'), {
+  ssr: false,
+});
+
+export async function getServerSideProps({ params, query, req, res }) {
+  const categoryDetails = await getCategoryDetails(params.categoryId);
+  let user = await isAuthenticated(req, res);
```
[the remark this answers is not in the document]
Comment (jamessspanggg) [reply pages/donations/category/[categoryId].js]: Yes have done so

File: src/components/category/pages/ViewAllDonationsPage.js [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+  const getLastQueriedDocument = () => {
+    if (allDonations.length === 0) {
+      return null;
+    }
+    return allDonations[allDonations.length - 1];
+  };
+
+  const getNextBatchOfDonations = async (filter, lastQueriedDocument) => {
+    // only time stamp should be reversed as it is from newest to oldest
+    const isReverse = filter === DonationsSortTypeConstant.TIMESTAMP ? true : false;
```
[the remark this answers is not in the document]
Comment (jamessspanggg) [reply src/components/category/pages/ViewAllDonationsPage.js:92]: We can add the `or` clause to this variable in the future. Or should I add it in this PR?

**Comment by jamessspanggg** [comment]
> enerally ok. Should we combine `ViewAllDonationsPage` & `DonationsViewCategoryPage`. Cause I see that `ViewAllDonationsPage` is quite similar to `DonationsViewCategoryPage`.

Currently the only difference is that the query being called is different & the category title of the `view all wishes page` is constant. I separated it to two components because I thought we might need different additions between the `view all wishes page` and `view a specific category of wishes` page in the future.

File: src/components/category/pages/ViewAllDonationsPage.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (jinyingtan) [reply src/components/category/pages/ViewAllDonationsPage.js:92]: ahh ok. then it's fine. we can add it when we have the distance

**Pull request merged**